hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -0,0 +1,869 @@
+	PreserveDeploymentAnnotation = "x-preserve-deployment"
+)
+
+// TODO: Do we actually want this at the controller level or this level?
+const GenericControllerFinalizer = "generated.infra.azure.com/finalizer"
+
+type CreateOrUpdateAction string
+
+const (
+	CreateOrUpdateActionNoAction          = CreateOrUpdateAction("NoAction")
+	CreateOrUpdateActionManageOwnership   = CreateOrUpdateAction("ManageOwnership")
+	CreateOrUpdateActionBeginDeployment   = CreateOrUpdateAction("BeginDeployment")
+	CreateOrUpdateActionMonitorDeployment = CreateOrUpdateAction("MonitorDeployment")
+)
+
+type DeleteAction string
+
+const (
+	DeleteActionBeginDelete   = DeleteAction("BeginDelete")
+	DeleteActionMonitorDelete = DeleteAction("MonitorDelete")
+)
+
+// TODO: I'm pretty sure we can come up with a way to drop the action parameter here
+type CreateOrUpdateActionFunc = func(ctx context.Context, action CreateOrUpdateAction) (ctrl.Result, error)
+type DeleteActionFunc = func(ctx context.Context, action DeleteAction) (ctrl.Result, error)
+
+var _ genruntime.Reconciler = &AzureDeploymentReconciler{}
+
+type AzureDeploymentReconciler struct {
+	obj                  genruntime.MetaObject
+	log                  logr.Logger
+	recorder             record.EventRecorder
+	ARMClient            armclient.Applier
+	KubeClient           *kubeclient.Client
+	ResourceResolver     *armresourceresolver.Resolver
+	CreateDeploymentName func(obj metav1.Object) (string, error)
+}
+
+func NewAzureDeploymentReconciler(
+	metaObj genruntime.MetaObject,
+	log logr.Logger,
+	armClient armclient.Applier,
+	eventRecorder record.EventRecorder,
+	kubeClient *kubeclient.Client,
+	resourceResolver *armresourceresolver.Resolver,
+	createDeploymentName func(obj metav1.Object) (string, error)) genruntime.Reconciler {
+
+	return &AzureDeploymentReconciler{
+		obj:                  metaObj,
+		log:                  log,
+		recorder:             eventRecorder,
+		ARMClient:            armClient,
+		CreateDeploymentName: createDeploymentName,
+		KubeClient:           kubeClient,
+		ResourceResolver:     resourceResolver,
+	}
+}
+
+func (r *AzureDeploymentReconciler) CreateOrUpdate(ctx context.Context) (ctrl.Result, error) {
+	action, actionFunc, err := r.DetermineCreateOrUpdateAction()
+
+	if err != nil {
+		r.log.Error(err, "error determining create or update action")
+		r.recorder.Event(r.obj, v1.EventTypeWarning, "DetermineCreateOrUpdateActionError", err.Error())
+		return ctrl.Result{}, err
+	}
+
+	result, err := actionFunc(ctx, action)
+	if err != nil {
+		r.log.Error(err, "Error during CreateOrUpdate", "action", action)
+		r.recorder.Event(r.obj, v1.EventTypeWarning, "CreateOrUpdateActionError", err.Error())
+		return ctrl.Result{}, err
+	}
+
+	return result, nil
+}
+
+func (r *AzureDeploymentReconciler) Delete(ctx context.Context) (ctrl.Result, error) {
+	action, actionFunc, err := r.DetermineDeleteAction()
```

Comment:
Ditto here - I think we can drop the `action` parameter for the delete action.

Reply:
Fixed